src/FakeItEasy/Creation/CastleDynamicProxy/CastleDynamicProxyGenerator.cs

```diff
@@ -40,18 +40,42 @@ public CastleDynamicProxyGenerator(CastleDynamicProxyInterceptionValidator inter
             IFakeCallProcessorProvider fakeCallProcessorProvider)
         {
             Guard.AgainstNull(customAttributeBuilders, "customAttributeBuilders");
-            Guard.AgainstNull(fakeCallProcessorProvider, "fakeCallProcessorProvider");
 
-            ProxyGenerationOptions.AdditionalAttributes.Clear();
+            var options = CreateProxyGenerationOptions();
             foreach (CustomAttributeBuilder builder in customAttributeBuilders)
             {
-                ProxyGenerationOptions.AdditionalAttributes.Add(builder);
+                options.AdditionalAttributes.Add(builder);
             }
 
-            return this.GenerateProxy(typeOfProxy, additionalInterfacesToImplement, argumentsForConstructor, fakeCallProcessorProvider);
+            return GenerateProxy(typeOfProxy, options, additionalInterfacesToImplement, argumentsForConstructor, fakeCallProcessorProvider);
+        }
+
+        public ProxyGeneratorResult GenerateProxy(
+            Type typeOfProxy,
+            IEnumerable<Type> additionalInterfacesToImplement,
+            IEnumerable<object> argumentsForConstructor,
+            IFakeCallProcessorProvider fakeCallProcessorProvider)
+        {
+            var options = CreateProxyGenerationOptions();
+            return GenerateProxy(typeOfProxy, options, additionalInterfacesToImplement, argumentsForConstructor, fakeCallProcessorProvider);
         }
 
-        public ProxyGeneratorResult GenerateProxy(Type typeOfProxy, IEnumerable<Type> additionalInterfacesToImplement, IEnumerable<object> argumentsForConstructor, IFakeCallProcessorProvider fakeCallProcessorProvider)
+        public bool MethodCanBeInterceptedOnInstance(MethodInfo method, object callTarget, out string failReason)
+        {
+            return this.interceptionValidator.MethodCanBeInterceptedOnInstance(method, callTarget, out failReason);
+        }
+
+        private static ProxyGenerationOptions CreateProxyGenerationOptions()
+        {
+            return new ProxyGenerationOptions(ProxyGenerationHook);
+        }

```

Comment:
We could, but why? I created a method because it was used in two places; if, for some reason or another, we need to make changes to the `ProxyGenerationOptions`, there will be only one place to change.